packages/clerk-js/src/core/clerk.ts

```diff
@@ -708,8 +708,16 @@ export class Clerk implements ClerkInterface {
     // to keep the organization id that the session had.
     const shouldSwitchOrganization = organization !== undefined;
     if (newSession && shouldSwitchOrganization) {
-      const organizationId = typeof organization === 'string' ? organization : organization?.id;
-      newSession.lastActiveOrganizationId = organizationId || null;
+      const organizationIdOrSlug = typeof organization === 'string' ? organization : organization?.id;
+
+      if (organizationIdOrSlug?.startsWith('org_')) {
+        newSession.lastActiveOrganizationId = organizationIdOrSlug || null;
+      } else {
+        const matchingOrganization = session?.user?.organizationMemberships.find(
+          mem => mem.organization.slug === organizationIdOrSlug,
+        );
+        newSession.lastActiveOrganizationId = matchingOrganization?.id || null;
```

Comment:
Get the `organization_id` by checking the memberships array using the string value passed which is likely a `slug`